src/spikeinterface/core/numpyextractors.py

```diff
@@ -85,7 +85,7 @@ def __init__(self, traces_list, sampling_frequency, t_starts=None, channel_ids=N
         }
 
     @staticmethod
-    def from_recording(source_recording, **job_kwargs):
+    def from_recording(source_recording, t_starts=None, **job_kwargs):
```

Comment:
Not sure to agree with this. The t_tsarts should be taken from the source recording. No ?

Reply:
Mayybe we should make a private method for this
https://github.com/SpikeInterface/spikeinterface/pull/3120/files#diff-6b93c507ba0be7d51533ba5d6d6bb00e9b4481bd887646d4e436496164ba0ec8R503-R508

and use it at several places. No ?

Reply:
done in last commit